components/supervisor/pkg/supervisor/git.go

```diff
@@ -42,15 +48,31 @@ func (p *GitTokenProvider) GetToken(ctx context.Context, req *api.GetTokenReques
 	for _, scp := range token.Scopes {
 		scopes[scp] = struct{}{}
 	}
+	missing := getMissingScopes(req.Scope, scopes)
+	if len(missing) > 0 {
+		message := fmt.Sprintf("An operation requires additional permissions: %s. Please grant these on the [access control page](%s).", strings.Join(missing, ", "), p.workspaceConfig.GitpodHost+"/access-control")
+		p.notificationService.Notify(ctx, &api.NotifyRequest{
```

Comment:
missing error handling

```suggestion
		_, err := p.notificationService.Notify(ctx, &api.NotifyRequest{
		    ...
		}
		if err != nil {
		    return nil, err
		}
```

Reply:
Error reporting the error :-)